runtime/bcverify/bcverify.c

```diff
@@ -954,8 +954,14 @@ mergeStacks (J9BytecodeVerificationData * verifyData, UDATA target)
 					/* Merge when either the source or target not an object */
 					if ((sourceItem | targetItem) & (BCV_BASE_OR_SPECIAL)) {
 
-						/* Mismatch results in undefined local - rewalk if modified stack */
-						if (*targetStackPtr != (UDATA) (BCV_BASE_TYPE_TOP)) {
+						/* Mismatch results in undefined local - rewalk if modified stack
+						 * Note: BCV_SPECIAL (specifically BCV_SPECIAL_INIT) must be reserved to flag the uninitialized_this object
+						 * existing in the stackmap frame when invoking setInitializedThisStatus()
+						 * after the stackmaps is successfully built.
+						 */
+						if ((*targetStackPtr != (UDATA) (BCV_BASE_TYPE_TOP))
+						&& !(*targetStackPtr & BCV_SPECIAL)
```

Comment:
> 
> 
> Isn't the local `targetItem` already initialized to be `*targetStackPtr`? Can the local be used here? Also, please use explicit boolean conditions

Yes, we should use `targetItem` instead of `*targetStackPtr` in this case. Updated as suggested.

Reply:
> Is this really checking for `(targetItem & BCV_SPECIAL_INIT) != 0` to avoid changing to `BCV_BASE_TYPE_TOP` in that case?

Yes, it was based on the thorough debugging with the class files in question.  It is obvious that the intention of the original implementation was to reserve the `BCV_SPECIAL` bits so as to set up `uninitializedThis ` in `setInitializedThisStatus` (the  `BCV_SPECIAL` bits can't be restored back or reset up later in `simulateStack` once they were removed here) but somehow the bits were wipe out here accidentally to deal with other situations. That's why we need to handle the case on both `mergeStack() ` and `matchStack()` to ensure `BCV_SPECIAL` bits are correctly handled.